public/docs/ts/latest/guide/testing.jade

```diff
@@ -1,102 +1,1758 @@
+  They should fake such calls. The setup in this `app/shared/twain.component.spec.ts` shows one way to do that: 
++makeExample('testing/ts/app/shared/twain.component.spec.ts', 'setup', 'app/shared/twain.component.spec.ts (setup)')(format='.')
+:marked
+  This is similar to the [`welcome.component.spec` setup](#welcome-spec-setup).
+  But instead of creating a fake service object, this setup injects the _real_ service (see the test module `providers`) and 
+  replaces the critical `getQuote` method with a Jasmine spy.
++makeExample('testing/ts/app/shared/twain.component.spec.ts', 'spy')(format='.')
+:marked
+  The spy is designed such that any call to `getQuote` receives an immediately resolved promise with a test quote.
+  The spy bypasses the actual `getQuote` method and therefore will not contact the server.
+
+  Here are the tests with commentary to follow:
++makeExample('testing/ts/app/shared/twain.component.spec.ts', 'tests', 'app/shared/twain.component.spec.ts (tests)')
+:marked
+  ### Synchronous tests
+  The first two tests are synchronous. 
+  Neither test can prove that a value from the service will be displayed.
+  
+  Thanks to the spy, the second test verifies that `getQuote` is called. 
+  But the quote itself has not arrived, despite the fact that the spy returns a resolved promise.
+  
+  This test must wait at least one full turn of the JavaScript engine, a least one "tick", before the
+  value becomes available. By that time, the test runner has moved on to the next test in the suite.
+
+  The test must become an "async test" ... like the third test
+
+a#async-fn-in-it
+:marked
+  ## The _async_ function in _it_
+
+  Notice the `async` in the third test.
++makeExample('testing/ts/app/shared/twain.component.spec.ts', 'async-test', 'app/shared/twain.component.spec.ts (async test)')(format='.')
+:marked
+  The `async` function is part of the _Angular TestBed_ feature set.
+  It _takes_ a parameterless function and _returns_ a parameterless function
+  which becomes the argument to the  Jasmine `it` call. 
+
+  The body of the `async` argument looks much like the body of a normal `it` argument.
+  There is nothing obviously asynchronous about it. For example, it doesn't return a promise.
+
+  The `async` function arranges for the tester's code to run in a special _async test zone_
+  that almost hides the mechanics of asynchronous execution.
+
+  Almost but not completely.
+
+a#when-stable
+:marked
+  ## _whenStable_
+  The test must wait for the `getQuote` promise to resolve. 
+ 
+  The `getQuote` promise promise resolves in the next turn of the JavaScript engine, thanks to the spy.
+  But a different test implementation of  `getQuote` could take longer.
+  An integration test might call the _real_ `getQuote`, resulting in an XHR request
+  that took many seconds to respond.
+
+  This test has no direct access to the `getQuote` promise
+  which is is private and inaccessible inside `TwainComponent`.
```

Comment:
`TwainComponent`:
```
  ngOnInit(): void {
    this.twainService.getQuote().then(quote => this.quote = quote);
  }
```
There is no obvious way to get the promise returned by calling `twainService.getQuote` inside the component. Is it really ambiguous?

Will get rid of double "is is" and add a bit of clarity